src/relations/osm_mysql.py

```diff
@@ -112,7 +112,7 @@ def _on_osm_mysql_relation_created(self, event: RelationCreatedEvent) -> None:
 
         # Ensure that config values exist, else we will be unable to create database and user
         if not username or not database:
-            self.unit.status = BlockedStatus("Missing `osm-mysql` relation data")
+            self.charm.unit.status = BlockedStatus("Missing `osm-mysql` relation data")
```

Comment:
JFMI, when do we set "self.unit.status" and when "self.charm.unit.status"?

```
> rg -c  self.unit.status tests/
tests/integration/relations/application-charm/src/charm.py:5
tests/integration/relations/application-charm/lib/charms/data_platform_libs/v0/database_requires.py:1
tests/integration/high_availability/application_charm/src/charm.py:2
tests/integration/high_availability/application_charm/lib/charms/data_platform_libs/v0/database_requires.py:1
```

```
> rg -c  self.charm.unit.status tests/
tests/unit/test_charm.py:4
```

Reply:
self.unit.status -> in the charm.py file (within the charm object itself)
self.charm.unit.status -> in files encapsulating relations (we pass in a reference of the charm object to the relation class [e.g. here](https://github.com/canonical/mysql-k8s-operator/blob/27297f80718dd62c4061f87d885d5f07d711b4d3/src/relations/mysql_provider.py#L47-L50))

Reply:
OK. It matches my understanding of the code, but is a great opportunity for the human mistakes.
Here is a grep for our repos:

Total uses of "self.unit.status":
```
10:30:31 ✔ taurus:~/canonical$ rg "self.unit.status" | wc -l
463
```

The "self.unit.status" used outside "charm.py:"
```
10:31:25 ✔ taurus:~/canonical$ rg "self.unit.status" | grep -v "charm.py:" | wc -l
141
```
Which I suspect is wrong is some/most cases.

In the same time we set "self.charm.unit.status" inside charm.py as well:
```
10:31:33 ✔ taurus:~/canonical$ rg "self.charm.unit.status" | grep "charm.py:" | wc -l
18
```

So, should we create a ticket to check it precisely?
IMHO, we should find a way to use one approach only and create an source-code automated test to catch human mistakes.

P.S. the full list of repos matched above: https://pastebin.canonical.com/p/725Y2MKXPZ/

Reply:
I would agree that a consistent approach would lead to less typos/bugs. Since the relation files (like [this](https://github.com/canonical/mysql-operator/blob/main/src/relations/mysql.py#L24)) inherits from `ops.framework.Object`, I believe we can use `self.unit.status` instead of `self.charm.unit.status`